jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/Content.java

```diff
@@ -477,6 +442,50 @@ static Chunk from(ByteBuffer byteBuffer, boolean last, Consumer<ByteBuffer> rele
             return new ByteBufferChunk.ReleasedByConsumer(byteBuffer, last, Objects.requireNonNull(releaser));
         }
 
+        /**
+         * <p>Creates a last/non-last Chunk with the given ByteBuffer, linked to the given Retainable.</p>
+         * <p>The {@link #retain()} and {@link #release()} methods of this Chunk will delegate to the
+         * given Retainable.</p>
+         *
+         * @param byteBuffer the ByteBuffer with the bytes of this Chunk
+         * @param last whether the Chunk is the last one
+         * @param retainable the Retainable this Chunk links to
+         * @return a new Chunk
+         */
+        static Chunk from(ByteBuffer byteBuffer, boolean last, Retainable retainable)
+        {
+            return new ByteBufferChunk.ReleasedByRetainable(byteBuffer, last, Objects.requireNonNull(retainable));
+        }
```

Comment:
The javadoc on all the from methods needs to be more explicit that there is an implicit retain already done on the returned Chunk, so that a single release will do the releasing, unless retain has been called.

Reply:
Like for `Source.read()` I think it's better to specify that a `release()` must be done, rather than going into details about `retain()`. For example, for this particular method, there is no `retain()` as the `Chunk` just wraps the given `Retainable`.